TodoListSPA/app.js

```diff
@@ -0,0 +1,175 @@
+'use strict';
+
+config.callback = loggedin;
+var authenticationContext = new AuthenticationContext(config);
+
+if (!config.popUp) {
+    if (authenticationContext.isCallback(window.location.hash)) {
+        authenticationContext.handleWindowCallback();
+    }
+    else {
+        var user = authenticationContext.getCachedUser();
+        if (user && window.parent === window && !window.opener) {
+            // Display the user
+            displayUserAndShowSignOutButton(user);
+
+            // Call the protected API to show the content of the todo list
+            acquireTokenAndCallService();
```

Comment:
I would rename this method as it sounds like it's creating an abstraction around ADAL that isn't necessary. Don't want to confuse a developer with usage of "acquireToken"